src/inference/gibbs.jl

```diff
@@ -28,23 +28,24 @@ mutable struct Gibbs{A} <: InferenceAlgorithm
     n_iters   ::  Int     # number of Gibbs iterations
     algs      ::  A   # component sampling algorithms
     thin      ::  Bool    # if thinning to output only after a whole Gibbs sweep
-    gid       ::  Int
 end
-Gibbs(n_iters::Int, algs...; thin=true) = Gibbs(n_iters, algs, thin, 0)
-Gibbs(alg::Gibbs, new_gid) = Gibbs(alg.n_iters, alg.algs, alg.thin, new_gid)
+Gibbs(n_iters::Int, algs...; thin=true) = Gibbs(n_iters, algs, thin)
 
 const GibbsComponent = Union{Hamiltonian,MH,PG}
 
 function Sampler(alg::Gibbs, model::Model)
+    info = Dict{Symbol, Any}()
+    spl = Sampler(alg, info)
+
     n_samplers = length(alg.algs)
     samplers = Array{Sampler}(undef, n_samplers)
-
     space = Set{Symbol}()
 
     for i in 1:n_samplers
         sub_alg = alg.algs[i]
         if isa(sub_alg, GibbsComponent)
-            samplers[i] = Sampler(typeof(sub_alg)(sub_alg, i), model)
+            samplers[i] = Sampler(sub_alg, model)
+            samplers[i].selector.tag[] = Symbol(typeof(sub_alg))
```

Comment:
This and the equivalent in PMMH seem to be the only 2 places where `tag` is modified after construction of the sampler. Can we not pass the tag to the constructor, and avoid the use of `Ref` entirely for performance?

Reply:
I agree that the use of `Ref` might be unnecessary if we avoiding mutating `selector`s after they are created. For this case, we can directly pass in the `tag` when constructing `samper`s, e.g.

```julia
samplers[i] = Sampler(sub_alg, model, Selector(Symbol(typeof(sub_alg))))
```

Reply:
Fixed by #726.